packages/fxa-settings/src/components/FormVerifyCode/index.tsx

```diff
@@ -84,8 +85,10 @@ const FormVerifyCode = ({
     localizedDefaultCodeRequiredMessage,
   ]);
 
-  const onSubmit = ({ code }: FormData) => {
-    verifyCode(code.trim());
+  const onSubmit = async ({ code }: FormData) => {
+    setIsSubmitting(true);
+    await verifyCode(code.trim());
```

Comment:
If this fails, do we end up stuck in an submitting state? (ie maybe there should be  try catch here?)

Reply:
This should be caught in the `onSubmit` `try`/`catch` in SigninTokenCode: https://github.com/mozilla/fxa/blob/11678ecec48985b6f306d186e3c0474a49e6d830/packages/fxa-settings/src/pages/Signin/SigninTokenCode/index.tsx#L120-L121

Looks like we're always wrapping the `try`/`catch` similarly when using `VerifyShortCode` in other components. I wonder if there might be an opportunity for some cleanup with FXA-7888, where we need to basically remake this component into a numbered input. If there's certain kinds of error messages we always want to display in the tooltip for example, maybe we should manage the try/catch and tooltip in the component.